SWIPs/swip-adaptive-node-protocol.md

```diff
@@ -0,0 +1,131 @@
+---
+SWIP: 12
+title: Adaptive nodes capabilities protocol
+author: Louis Holbrook <dev@holbrook.no> (https://holbrook.no), Tim Bansemer (editorial) @FantasticoFox <tim@ethswarm.org>
+status: Draft
+type: Track Networking
+created: 2019-07-11
+---
+
+## Abstract
+
+Adaptive node operation means a node should be able to change the capabilities it has according to changes in available resources.
+
+## Motivation
+
+The peers of the node may need to adjust their behavior towards the node when these capabilities change. Therefore, a protocol is required to inform the peers of changes in capabilities.
+
+## Specification
```

Comment:
The first part reads more like an introduction, perhaps it can be under such a header? Then more clear definitions under Definitions header, Flow, Wire/data layout etc